src/IonWriter.ts

```diff
@@ -17,43 +17,138 @@ import {Reader} from "./IonReader";
+
+    /**
+     * Writes a [[Decimal]] value as an Ion `decimal` value.
+     *
+     * @param value The [[Decimal]] to write, which may be `null` to write a `null.decimal`.
+     */
+    writeDecimal(value: Decimal | null) : void;
+
+    writeTimestamp(value: Timestamp) : void;
+
+    /**
+     * Writes a `string` value as an Ion `string`.
+     *
+     * @param value The `string` to write, which may be `null` to write a `null.string`.
+     */
+    writeString(value: string | null) : void;
+
+    /**
+     * Writes a `string` value as an Ion `symbol`.
+     *
+     * @param value The `string` to write, which may be `null` to write a `null.symbol`.
+     */
+    writeSymbol(value: string | null) : void;
+
+    /**
+     * Writes a `Uint8Array` as an Ion `blob` value.
+     *
+     * @param value The array to write, which may be `null` to write a `null.blob`.
+     */
+    writeBlob(value: Uint8Array | null) : void;
+
+    /**
+     * Writes a `Uint8Array` as an Ion `clob` value.
+     *
+     * @param value The array to write, which may be `null` to write a `null.clob`.
+     */
+    writeClob(value: Uint8Array | null) : void;
+
+    /**
+     * Writes a reader's current value and all following values until the end
+     * of the current container.  If there's no current value then this method
+     * calls {@link next()} to get started.
+     */
+    writeValues(reader: Reader) : void;
+
+    /**
+     * Writes the field name for a member of a `struct`.
+     *
+     * @throws Error if the [[Writer]] is not within a `struct` or this method was already
+     *  called before a value was written.
+     */
+    writeFieldName(fieldName: string) : void;
+
+    /**
+     * Starts a container and positions the writer within that container.
+     *
+     * @throws Error if `type.container` is not one of `IonTypes.LIST`, `IonTypes.SEXP`,
+     *  or `IonTypes.STRUCT`.
+     */
+    stepIn(type: IonType) : void;
+
+    /**
+     * Steps out of a container and positions the writer after the container.
+     *
+     * @throws Error if the writer is not inside a container.
+     */
+    stepOut() : void;
+
+    /**
+     * Adds an annotation to the list of annotations to be used when
+     * writing the next value.
+     */
+    addAnnotation(annotation: string) : void;
+
+    /**
+     * Specifies the list of annotations to be used when writing
+     * the next value.  This clears the current annotations set
+     * for the next value.
+     */
+    setAnnotations(annotations: string[]) : void;
+
+    /**
+     * Flushes data to the internal buffer and finalizes the Ion stream.
+     *
+     * @throws Error if the [[Writer]] is already closed or the writer is not
+     *  at the top-level (i.e. inside of a container).
+     */
+    close() : void;
+
+    /**
+     * Retrieves the serialized buffer as an array of octets.  Generally this will
+     * be either a UTF-8 encoded buffer for Ion text or Ion binary.  The buffer is
```

Comment:
> Generally this will be either a UTF-8 encoded buffer for Ion text or Ion binary.

* You say "generally" -- are there other options?
* This was a bit of a garden path sentence for me. I would rearrange it a bit for easier parsing:
   > Generally, this will be either Ion binary or a UTF-8 encoded buffer for Ion text.

   Alternatively, since we already know the return type to be `Uint8Array`:
   > [The buffer] will contain either Ion binary data or UTF-8 encoded Ion text.

Reply:
Good point, I did think it wasnkind of hedgy as I wrote it, let me reword.

The use case I was thinking wss an Ion writer for things that don't serialize (e.g. to a DOM) or using the interface to leverage the logical Ion types but emit something like CBOR.

The API is not amenable to the former, and the later is somewhat a stretch.

Reply:
Fixed